mempool/mempool.go

```diff
@@ -77,6 +77,10 @@ type Config struct {
 	// difficulty for the block after the current best block.
 	//
 	// This function must be safe for concurrent access.
+	//
+	// TODO: Ideally this would not return an error, but changing the function
+	// would require a major version bump to the mempool module.
+	// The next time a major version is released, the error should be removed.
```

Comment:
While this is true for the particular implementation currently used in this PR, the point of decoupling the structures by using a function reference is that you don't need to know whether the impl returns an error or not. So I'd say that you should just maintain this function signature and the subsequent error checking.